src/test/java/test/order/OrderTest.java

```diff
@@ -18,13 +18,11 @@ public void interleavingMethodsInDifferentClasses() {
     assertThat(listener.getSucceedMethodNames()).containsExactly(
         "beforeClass",
         "test1",
-        // "test3", // The result is not exactly the one from #288 (prior 6.8)
-        // but it respects the class separation
+        "test3",
```

Comment:
From what I can tell from the comment that used to be here, it doesn't make a difference whether test3 is run before test4 or after.

Reply:
Actually, I've looked further into this, and I think that according to everything else I've seen about test order in TestNG, `test3` should be running before `test4`, because `test3` comes first alphabetically and they have equal priority. The reason `test4` gets run first is because when TestNG runs single-threaded, nodes aren't marked as`Status.FINISHED` and `freeNodes` isn't re-calculated until after the prior "batch" of `freeNodes` has been completed (this is in the non-parallel branch of `TestRunner#privateRun`). Because `test3` has a dependency on `test1`, it wasn't included in the first "batch", but `test4` was.